docker/local-pytest/Dockerfile

```diff
@@ -27,7 +27,36 @@ RUN apt-get install -yq \
             postgresql-plpython-9.5 \
             postgresql-9.5-postgis-2.4 \
             && \
-    pip3 install -U pip pytest
+    pip3 install -U pipenv
+
+# Build arguments (scope limited to build). If you wish to use a different user,
+# group, or home dir, override these in the build command or change them here.
+# If you specify build arg USERNAME=root, then the user is root.
+ARG USERNAME=dockremap
+ARG UID=1000
+ARG GROUPNAME=${USERNAME}
+ARG GID=1000
+ARG USER_DIR=/opt/${USERNAME}
+
+# Environment variables (scope NOT limited to build). These are set here so that
+# subsequent builds and containers have access to these build arguments.
+ENV USERNAME=${USERNAME}
+ENV UID=${UID}
+ENV GROUPNAME=${GROUPNAME}
+ENV GID=${GID}
+ENV USER_DIR=${USER_DIR}
+
+# Create non-privileged user, group, and its directory. This is only done if USERNAME is not root.
+RUN if [ "$USERNAME" != "root" ]; \
+    then \
+        echo "Creating non-root user"; \
+        groupadd -r -g ${GID} ${GROUPNAME}; \
+        useradd -r -d ${USER_DIR} -g ${GROUPNAME} -u ${UID} ${USERNAME}; \
+        mkdir -p ${USER_DIR}; \
+        chown ${USERNAME}:${GROUPNAME} ${USER_DIR}; \
+    fi
+
+USER ${USERNAME}
```

Comment:
Does this package require that we are not running as root for its tests? Or has this been implemented for flexibility?

Reply:
Two reasons:
- The original setup changed to a non-root user (in `entrypoint.sh`) because Postgres refuses to run as root for security reasons. 
- In general, we do not want to be running our test environments as root on our workstations, because they are behind the firewall. This follows the pattern established in several other projects for non-root users. I'd very much like to factor it out so that there isn't so much code repetition, but time ...